### PR TITLE
New sample: setup-location-driven-geotriggers

### PR DESCRIPTION
**Pull request by shubham7109**
PR to add Geotriggers to the Android Sample Viewer.

### CODE REVIEW

File: kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+                        // create a new garden section data object and add it to the hash map of visited sections
+                        sectionsVisited[fenceFeatureName] =
+                            GardenSection(
+                                fenceFeature,
+                                fenceFeatureName,
+                                description,
+                                attachmentImageURI
+                            )
+                        populateUI(
+                            sectionsVisited[fenceFeatureName]!!,
```
Comment (TADraeseke) [inline kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt:284]: let's add a null check in the populateUI method and avoid using !!s if we can

File: kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+            // Use the compress method on the BitMap object to write image to the OutputStream
+            val bitmapImage = BitmapFactory.decodeStream(imageInputStream)
+            bitmapImage.compress(Bitmap.CompressFormat.PNG, 100, fileOutputStream)
+        } catch (e: Exception) {
+            e.printStackTrace()
+        } finally {
+            try {
+                fileOutputStream?.close()
+            } catch (e: IOException) {
+                e.printStackTrace()
```
Comment (TADraeseke) [inline kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt:360]: I know this is my old code, but let's handle the stackTrace's by writing the message the Log.e

File: kotlin/setup-location-driven-geotriggers/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+            <androidx.constraintlayout.widget.ConstraintLayout
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content">
+
+                <TextView
+                    android:layout_width="0dp"
+                    android:layout_height="wrap_content"
+                    android:layout_margin="8dp"
+                    android:text="Current Garden Section:"
```
Comment (TADraeseke) [inline kotlin/setup-location-driven-geotriggers/src/main/res/layout/activity_main.xml:38]: let's put all these into R.string

File: kotlin/setup-location-driven-geotriggers/src/main/res/layout/activity_main.xml [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                    android:textColor="@color/white"
+                    app:layout_constraintBottom_toBottomOf="parent"
+                    app:layout_constraintEnd_toEndOf="parent"
+                    app:layout_constraintTop_toTopOf="parent" />
+
+            </androidx.constraintlayout.widget.ConstraintLayout>
+
+            <View
+                android:layout_width="match_parent"
+                android:layout_height="2px"
```
Comment (TADraeseke) [inline kotlin/setup-location-driven-geotriggers/src/main/res/layout/activity_main.xml:66]:
```suggestion
                android:layout_height="2dp"
```

File: kotlin/setup-location-driven-geotriggers/src/main/res/layout/list_item.xml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+    <Button
+        android:id="@+id/itemButton"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="center_horizontal"
+        android:background="@drawable/rounded_button_secondary"
+        android:paddingStart="10dp"
+        android:paddingEnd="10dp"
+        android:text="Test"
```
Comment (TADraeseke) [inline kotlin/setup-location-driven-geotriggers/src/main/res/layout/list_item.xml:15]: this should be removed

File: kotlin/setup-location-driven-geotriggers/src/main/res/layout/activity_main.xml [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+        </LinearLayout>
+
+    </androidx.core.widget.NestedScrollView>
+
+    <com.google.android.material.floatingactionbutton.FloatingActionButton
+        android:id="@+id/playPauseFAB"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginEnd="30dp"
```
Comment (TADraeseke) [inline kotlin/setup-location-driven-geotriggers/src/main/res/layout/activity_main.xml:99]: add contentDescription

File: kotlin/setup-location-driven-geotriggers/src/main/res/layout/dialog_fragment.xml [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        android:layout_marginEnd="16dp"
+        android:padding="10dp"
+        android:textSize="18sp"
+        android:textStyle="bold" />
+
+    <ImageView
+        android:id="@+id/gardenContentImageView"
+        android:layout_width="190dp"
+        android:layout_height="120dp"
+        android:layout_gravity="center_horizontal"
```
Comment (TADraeseke) [inline kotlin/setup-location-driven-geotriggers/src/main/res/layout/dialog_fragment.xml:22]: add contentDescription

**Review by TADraeseke — CHANGES_REQUESTED** [review]
Thanks @shubham7109 -- just some comments in line.

I like the play/pause functionality.

As to the top UI. Is there a reason the Points of interest areas is so much larger than the Current Garden Section (let's make those section titles sentence case)? Can they both be the same size?

The N/A doesn't seem very useful for Points of interest. Maybe the UI should be smaller when there aren't any or even disappear entirely?

File: kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt (hunk 1 above)
Comment (shubham7109) [reply kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt:284]: Done!

**Comment by shubham7109** [comment]
- ImageView is now retrieved when the description dialog is clicked, hence now there is no delay when displaying the `GardenSection` button. 
- ListView still does not display multiple `GardenSections`. Using `Log.e` and during debug the `ListView` contains multiple items but does not render to the view. Could you take a look at this? @TADraeseke

File: kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt [hunk 8]
```diff
@@ -0,0 +1,400 @@
...
+                }
+            } else {
+                poiList.add(gardenSection)
+                poiListAdapter.notifyDataSetChanged()
+                Log.e("ADDED: ", gardenSection.title)
+                activityMainBinding.listAvailable.visibility = View.GONE
+            }
+        } else {
+            Toast.makeText(this, "Garden Section is null", Toast.LENGTH_SHORT).show()
+            Log.e("NullPointerException: ", "GardenSection is null")
```
Comment (TADraeseke) [inline kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt:332]: This isn't a property formed Log.e

File: kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt [hunk 9]
```diff
@@ -0,0 +1,400 @@
...
+                        GardenDescriptionFragment(gardenSection, this).show(
+                            supportFragmentManager,
+                            "GardenDescriptionFragment"
+                        )
+                    }
+                }
+            } else {
+                poiList.add(gardenSection)
+                poiListAdapter.notifyDataSetChanged()
+                Log.e("ADDED: ", gardenSection.title)
```
Comment (TADraeseke) [inline kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt:327]: These should all go now that we've gotten to the bottom of the problem with missing POIs

File: kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt [hunk 10]
```diff
@@ -0,0 +1,400 @@
...
+        setContentView(activityMainBinding.root)
+
+        // authentication with an API key or named user is required to access basemaps and other
+        // location services
+        ArcGISRuntimeEnvironment.setApiKey(BuildConfig.API_KEY)
+
+        val portal = Portal("https://www.arcgis.com", true)
+        // This sample uses a web map with a predefined tile basemap, feature styles, and labels
+        val map = ArcGISMap(PortalItem(portal, "6ab0e91dc39e478cae4f408e1a36a308"))
+        mapView.map = map
```
Comment (TADraeseke) [inline kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt:106]:
```suggestion
    mapView.map =  ArcGISMap(PortalItem(portal, "6ab0e91dc39e478cae4f408e1a36a308"))
```

File: kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt [hunk 11]
```diff
@@ -0,0 +1,400 @@
...
+
+    private val playPauseFAB: FloatingActionButton by lazy {
+        activityMainBinding.playPauseFAB
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(activityMainBinding.root)
+
+        // authentication with an API key or named user is required to access basemaps and other
```
Comment (TADraeseke) [inline kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt:99]:
```suggestion
        // Authentication with an API key or named user is required to access basemaps and other
```

File: kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt [hunk 12]
```diff
@@ -0,0 +1,400 @@
...
+                poiList.add(gardenSection)
+                poiListAdapter.notifyDataSetChanged()
+                Log.e("ADDED: ", gardenSection.title)
+                activityMainBinding.listAvailable.visibility = View.GONE
+            }
+        } else {
+            Toast.makeText(this, "Garden Section is null", Toast.LENGTH_SHORT).show()
+            Log.e("NullPointerException: ", "GardenSection is null")
+        }
+
```
Comment (TADraeseke) [inline kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt:334]: remove whitespace

File: kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt [hunk 13]
```diff
@@ -0,0 +1,400 @@
...
+            bitmapImage.compress(Bitmap.CompressFormat.PNG, 100, fileOutputStream)
+        } catch (e: Exception) {
+            Log.e("Exception: ", e.message.toString())
+            e.printStackTrace()
+        } finally {
+            try {
+                fileOutputStream?.close()
+            } catch (e: IOException) {
+                Log.e("IOException: ", e.message.toString())
+                e.printStackTrace()
```
Comment (TADraeseke) [inline kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt:380]: let's not print while stack trace if we're handling the error

File: kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt [hunk 14]
```diff
@@ -0,0 +1,400 @@
...
+            val bitmapImage = BitmapFactory.decodeStream(imageInputStream)
+            bitmapImage.compress(Bitmap.CompressFormat.PNG, 100, fileOutputStream)
+        } catch (e: Exception) {
+            Log.e("Exception: ", e.message.toString())
+            e.printStackTrace()
+        } finally {
+            try {
+                fileOutputStream?.close()
+            } catch (e: IOException) {
+                Log.e("IOException: ", e.message.toString())
```
Comment (TADraeseke) [inline kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt:379]: Check some of the other samples for formatting for Log statements. We usually use a TAG with the sample's name as the first argument

File: kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt [hunk 15]
```diff
@@ -0,0 +1,400 @@
...
+     * Initialize a simulation using a simulated data source and then
+     * feed it to the [LocationGeotriggerFeed]
+     */
+    private fun initializeSimulatedLocationDisplay(): LocationGeotriggerFeed {
+        val simulatedLocationDataSource = SimulatedLocationDataSource()
+
+        // Create SimulationParameters starting at the current time, a velocity of 10 m/s, and a horizontal and vertical accuracy of 0.0
+        val simulationParameters = SimulationParameters(Calendar.getInstance(), 3.0, 0.0, 0.0)
+
+        // Use the polyline as defined above or from this AGOL GeoJSON to define the path. retrieved
```
Comment (TADraeseke) [inline kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt:138]:
```suggestion
        // Use the polyline as defined above or from this ArcGIS Online GeoJSON to define the path. retrieved
```

File: kotlin/setup-location-driven-geotriggers/src/main/res/values/colors.xml [hunk 16]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <color name="colorPrimary">#2196F3</color>
+    <color name="colorPrimaryDark">#1976D2</color>
+    <color name="colorAccent">#4CAF50</color>
+
+    <color name="gardenPrimary">#3b4e1e</color>
+    <color name="gardenSecondary">#ac901e</color>
+    <color name="white">#FFFF</color>
+    <color name="semiTransparent">#DF000000</color>
```
Comment (TADraeseke) [inline kotlin/setup-location-driven-geotriggers/src/main/res/values/colors.xml:10]: remove unused resource

**Review by TADraeseke — CHANGES_REQUESTED** [review]
Just a few small things--behaviour of the app looks good, thanks @shubham7109!

**Pull request merged**